core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -72,7 +72,8 @@ private[spark] class ReliableCheckpointRDD[T: ClassTag](
       .sortBy(_.toString)
     // Fail fast if input files are invalid
     inputFiles.zipWithIndex.foreach { case (path, i) =>
-      if (!path.toString.endsWith(ReliableCheckpointRDD.checkpointFileName(i))) {
+      if (path.getName != ReliableCheckpointRDD.checkpointFileName(i) &
```

Comment:
For validating the file name `part-[0-9]+` is definitely concise, however we will loose the consistency in the understanding here. While validating we would be flexible but while actually getting the files we would always look for a 10digit formatting scheme. This would always fail if the part files don't match. So adding a `part-[0-9]+` check wont help.